### DISCUSSION TITLE
How to get uploaded data from Cloudinary?

### DISCUSSION THREAD

**Discussion by K-Juno** [neuefische/questions#234] [Web] [answered]
Hi everyone,

I have added an upload widget connected to my Cloudinary folder (thanks for the workshop, @pixelass !) in order to enable uploading audio files.
The upload itself is working just fine, but I can't figure out how to get those files in my App.
Do I need a connection to a database??
I tried to integrate a media library widget, but I just don't get the [documentation](https://cloudinary.com/documentation/media_library_widget#4_instantiate_and_open_the_media_library_widget).

All I want is to be able to play and delete those files again.

For now, I managed to get the output of selected/uploaded file in a media player (and a useless remove button, ignore that).

Thanks for any kind of hint!

My current PR: https://github.com/K-Juno/capstone-project/pull/15

**Comment by ThomasKirst** [comment, marked as answer]
Hi Juni,

I tried to upload a sound file on your preview URL.
Unfortunately, I got an error:

![Screenshot 2022-10-10 at 14 33 44](https://user-images.githubusercontent.com/4458383/194867219-5bbefa16-76c0-47c7-a373-97641181437f.png)

You might want to check your environment variables / settings on Vercel.
It seems as if the "upload preset" for cloudinary is missing.

Once the file is successfully uploaded, you can use the returned JSON data and store the `secure_url` of the response – which is the link to your new audio file – within your local state, in localStorage, or in your database.

Do you need the file attached to your lyrics model or are these uploads independent from the lyrics which are entered on another page?

For the part that removes an audio file it is important to know how you plan to store that state (again, local state / localStorage / database).

–Thomas

**Comment by K-Juno** [comment]
Hi Thomas,

OMG, you found an issue I wasn't even aware of!
Adding environment variables on Vercel: Check!
I will go on and try to figure out what's the best solution for accessing the data.
They are not linked in any way with the Lyrics section - not yet!
Thanks a lot!
